guava/src/com/google/common/math/IntMath.java

```diff
@@ -699,6 +699,29 @@ public static int mean(int x, int y) {
     // The alternative (x + y) >>> 1 fails for negative values.
     return (x & y) + ((x ^ y) >> 1);
   }
+  public static int digitsOfNFactorial(int n) {
```

Comment:
This is not a function I'm aware of anyone needing ever.  Can you discuss your use case?